src/services/CertificateGenerationService.ts

```diff
@@ -235,113 +235,53 @@ class CertificateGenerationService {
   /**
    * Determines if a test station is located in Wales
    * @param testStationPNumber The test station's P-number.
-   * @returns Promise<boolean> true if the test station is Welsh, false otherwise
+   * @returns Promise<boolean> true if the test station country is set to Wales, false otherwise
    */
   public async isTestStationWelsh(testStationPNumber: string): Promise<boolean> {
-    const testStations = await this.getTestStations();
-    const testStationPostcode = this.getThisTestStation(testStations, testStationPNumber);
-    return testStationPostcode ? await this.lookupPostcode(testStationPostcode) : false;
+    const testStation = await this.getTestStation(testStationPNumber);
+
+    if (!testStation.testStationPNumber) {
```

Comment:
Why are we checking a specific property? if the `getTestStation` method fails, are we still returning an object?

Reply:
As getTestStation method always returns an ITestStation object, this was a way of determining whether or not the expected values had been populated, and testStationPNumber 'should' always have been populated given that is what was used to retrieve the data from the endpoint in the first instance.

I just wanted to make sure we logged something in the event an object other than ITestStation was returned by the getTestStationByPNumber call.